CHANGES/.TEMPLATE.md

```diff
@@ -1,13 +1,16 @@
 
 {# TOWNCRIER TEMPLATE #}
 {% for section, _ in sections.items() %}
-{% if section %}### {{section}}
+{%- set section_slug = "-" + section|replace(" ", "-")|replace("_", "-")|lower %}
+{% if section %}### {{section}} {: #{{versiondata.version}}{{section_slug}} }
 
+{% else %}
+{%- set section_slug = "" %}
 {% endif %}
 
 {% if sections[section] %}
 {% for category, val in definitions.items() if category in sections[section]%}
-#### {{ definitions[category]['name'] }}
+#### {{ definitions[category]['name'] }} {: #{{versiondata.version}}{{section_slug}}-{{category}} }
```

Comment:
While being a good change, this may or may not lead to a one time hickup when collecting changelogs from previous branches.
At this time, just something to keep an eye on.